packages/analytics-core/src/core-client.ts

```diff
@@ -67,11 +73,18 @@ export class AmplitudeCore<T extends Config> implements CoreClient<T> {
   }
 
   async add(plugin: Plugin) {
-    const config = this.config;
-    return this.timeline.register(plugin, config);
+    if (!this.config) {
+      this.q.push(this.add.bind(this, plugin));
+      return;
+    }
+    return this.timeline.register(plugin, this.config);
   }
 
   async remove(pluginName: string) {
+    if (!this.config) {
+      this.q.push(this.remove.bind(this, pluginName));
+      return;
```

Comment:
Minor, do we want to return a similar resolved promise here, to be align with `this.timeline.deregister`? Same question for the early return in `add`.

Reply:
`return undefined` is same as `return Promise.resolve()` in async function. So it should be fine.

Reply:
That's correct. Unless @liuyang1520, you are referring to await until the plugins are actually added/removed?

Reply:
Got it, no worries, then, I forget that `async` will return a promise anyways.